### ISSUE TITLE
Rename ESM module output to .js (or add copy) for TypeScript compatibility

### ISSUE THREAD

**Issue by KeithHenry** [GoogleChromeLabs/comlink#378]
In TypeScript I can make a bare module reference:

```
import * as Comlink from '../lib-path/comlink/dist/esm/comlink';
```

But that adds the bare module to the output JS and requires an additional build step. I can usually work around this restriction with a `.js` extension:

```
import * as Comlink from '../lib-path/comlink/dist/esm/comlink.js';
```

Now the output JS gets the explicit `.js` path that works, and TS still knows to check `comlink.d.ts` for the definitions.

The problem is that Comlink uses `.mjs`, so I have to do:

```
import * as Comlink from '../lib-path/comlink/dist/esm/comlink.mjs';
```

TypeScript doesn't support this and it throws "Cannot find module" errors. It doesn't look like TypeScript will support `.mjs` any time soon (https://github.com/Microsoft/TypeScript/issues/27957 is a year old).

This means Comlink _cannot_ work with TypeScript without additional build/renaming tools.

I realise that this can be easily worked around with these tools, but it's an unnecessary barrier to using Comlink.

The `esm/dist` folder should include a `.js` and `.min.js` output, otherwise there's no point it having a `.d.ts` output in there either.

**Comment by surma** [comment]
That is a good catch! You are right. I'll add a `.js` file to `esm` :)

**Comment by surma** [comment]
Fixed in v4.0.5 :) Thanks for flagging!
